src/Neo/SmartContract/Native/NativeContract.cs

```diff
@@ -199,13 +200,21 @@ private NativeContractsCache.CacheEntry GetAllowedMethods(ProtocolSettings setti
         /// <param name="settings">The <see cref="ProtocolSettings"/> where the HardForks are configured.</param>
         /// <param name="index">Block index</param>
         /// <returns>The <see cref="ContractState"/>.</returns>
-        public ContractState GetContractState(ProtocolSettings settings, uint index)
+        public ContractState GetContractState(ProtocolSettings settings, uint index) => GetContractState(settings.IsHardforkEnabled, index);
+
+        /// <summary>
+        /// The <see cref="ContractState"/> of the native contract.
+        /// </summary>
+        /// <param name="hfChecker">Hardfork checker</param>
+        /// <param name="index">Block index</param>
+        /// <returns>The <see cref="ContractState"/>.</returns>
+        public ContractState GetContractState(IsHardforkEnabledDelegate hfChecker, uint index)
         {
             // Get allowed methods and nef script
-            NativeContractsCache.CacheEntry allowedMethods = GetAllowedMethods(settings, index);
+            var allowedMethods = GetAllowedMethods(hfChecker, index);
```

Comment:
Just a question from my side: `var`-style variable declaration is a kind of accepted standard for this repo now? E.g. should we declare variables in the same way (with type omission)?

Reply:
I prefer `Type x=new()` but @Jim8y and @cschuchardt88 prefer this style

Reply:
> Type x=new()

This way is more clear wrt variable types, and I see that a lot of old code support this way. OK.

Reply:
Well, I am not in any preference, I can accept any type, I just don't want to ban var. to be honest, I don't care about the style at all.